docsbox/docs/views.py

```diff
@@ -133,16 +154,24 @@ def get(self, task_id):
                             "mimeType": task.result["mimeType"],
                             "fileName": task.result["fileName"]
                         }
+                        app.logger.log(logging.INFO, response, extra={"request": request, "status": 200})
+                        return response
                     else:
-                        try:
-                            return send_from_directory(app.config["MEDIA_PATH"], task.id, as_attachment=True, attachment_filename=task.result["fileName"])
+                        try:                          
+                            response= send_from_directory(app.config["MEDIA_PATH"], task.id, as_attachment=True, attachment_filename=task.result["fileName"])
+                            app.logger.log(logging.INFO, "file: %s"%(task.result["fileName"]), extra={"request": request, "status": 200})
                         except exceptions.Timeout:
+                            app.errlog.log('CRITICAL', "VIA service took too long to respond.", extra={"request": request, "status": 504})
                             abort(504, "VIA service took too long to respond.")
+                    return response
                 else:
+                    app.errlog.log(logging.ERROR, "Task with no result", extra={"request": request, "status": 404})
                     abort(404, message="Task with no result")
             else:
+                app.errlog.log(logging.ERROR, "Task is still queued", extra={"request": request, "status": 400})
                 abort(400, message="Task is still queued")
         else:
+            app.errlog.log(logging.ERROR, "Unknown task_id", extra={"request": request, "status": 404})
```

Comment:
To handle a bit of the "spam" of these logs could we create our own abort() function, which would handle the error logging and after that call the flask's abort() function?

Reply:
```
def abort(error_code, message, request):
    if error_code >= 500:
        error_level = logging.CRITICAL
    elif error_code >= 400:
        error_level = logging.ERROR
    elif error_code >= 300:
        error_level = logging.WARNING
    app.errlog.log(error_level, message, extra={"request": request, "status": error_code})
    flask_restful.abort(error_code, message)
```

Is this good?

Reply:
`def abort(error_code, message, request):
    if error_code >= 500:
        error_level = logging.CRITICAL
    elif error_code >= 400:
        error_level = logging.ERROR
    elif error_code >= 300:
        error_level = logging.WARNING
    app.errlog.log(error_level, message, extra={"request": request, "status": error_code})
    flask_abort(error_code, message=message)`

Is this good? Or can it be improved ?